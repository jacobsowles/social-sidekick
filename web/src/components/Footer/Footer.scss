@import '@styles/_variables.scss';

.footer {
  background-color: $gray-dark;
  position: absolute;
  color: #fff;
  bottom: 0;
  width: 100%;
  height: 45px;
  line-height: 45px;

  a {
    color: #ffffff;

    &:hover {
      text-decoration: none;
    }
  }

  svg {
    font-size: 1.7em;
    margin-right: 10px;
    vertical-align: -0.25em;
  }
}
