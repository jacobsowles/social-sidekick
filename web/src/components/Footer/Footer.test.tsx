import React from 'react';
import {
  init,
  mountTopLevelElement,
  shallow,
  shallowTopLevelElement,
  toJson
} from '@tests/component-test-base';
import Footer, { FooterProps } from './Footer';

describe('Footer', () => {
  let props: FooterProps;

  init(
    () => (
      <Footer {...props}>
        <p>Test</p>
      </Footer>
    ),
    'footer'
  );

  const container = () => {
    return mountTopLevelElement().children();
  };

  beforeEach(() => {
    props = {
      className: undefined
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  describe('the rendered footer element', () => {
    it('should render a bootstrap container', () => {
      container()
        .prop('className')
        .should.equal('container');
    });
  });

  describe('when `className` is defined', () => {
    beforeEach(() => {
      props.className = 'another-class';
    });

    it('should add the class name to the footer', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.equal('footer another-class');
    });
  });

  describe('when `className` is not defined', () => {
    beforeEach(() => {
      props.className = undefined;
    });

    it('should only include the default class names', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.equal('footer');
    });
  });
});
