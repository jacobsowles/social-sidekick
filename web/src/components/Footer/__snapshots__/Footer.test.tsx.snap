// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the component without crashing 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <p>
      Test
    </p>
  </div>
</footer>
`;
